### ISSUE TITLE
Allow `.` in clientId fallback cookie

### ISSUE THREAD

**Issue by pdufour** [ampproject/amphtml#5729]
## Short description of your issue:

**TL;DR: Can `.` be added as an allowed character for clientIds?** (https://github.com/ampproject/amphtml/blob/master/extensions/amp-analytics/0.1/cid-impl.js#L115-L117)

Snowplow has an `amp-analytics` integration https://github.com/ampproject/amphtml/blob/master/extensions/amp-analytics/0.1/vendors.js#L811, that we are migrating to from JS tracker. But to retain session information correctly, we'd like to use the existing session ID cookie, which looks something like `_sp_id.44`. So we could theoretically do that by setting the `sid` value to `${clientID(_sp_id.44)}`. The only problem is that fails the client id validation, which only allows alphanumeric characters, and `-` and `_`.

## How do we reproduce the issue?

1. Create amp-analytics config with `${clientID(_sp_id.44)}` somewhere in one of the requests.
2. Get error (`The client id name must only use the characters...`)

## What browsers are affected?

All

## Which AMP version is affected?

All

**Comment by cramforce** [comment]
Yeah, no problem. @pdufour Could you send a PR that extends the regex here? https://github.com/ampproject/amphtml/blob/master/extensions/amp-analytics/0.1/cid-impl.js#L115
